### ISSUE TITLE
why's the function asnumpy() so slow?

### ISSUE THREAD

**Issue by cookingbear** [apache/incubator-mxnet#9317]
I tried the speech_recognition example but found out that the last step below(pred = pred.asnumpy()) slow down the process. why's that?

<pre>
`def update(self, labels, preds):
     check_label_shapes(labels, preds)
        if self.is_logging:
            log = LogUtil().getlogger()
            labelUtil = LabelUtil.getInstance()
        self.batch_loss = 0.
        for label, pred in zip(labels, preds):
            label = label.asnumpy()
            pred = pred.asnumpy()
`
</pre>

**Comment by sxjscience** [comment]
`asnumpy()` will trigger a cuda synchronization. So your code will be slowed down.

**Comment by vdantu** [comment]
@cookingbear : Does this answer your question? Since this is a general question, please move this discussion to discuss.mxnet.io to get a larger audience.

@sandeep-krishnamurthy : please label this issue as "Question" and close this.
